### DISCUSSION TITLE
Warning: annot type has no fill color

### DISCUSSION THREAD

**Discussion by rishkum** [pymupdf/PyMuPDF#1034] [Looking for help] [answered]
Hi, Thanks for this amazing library.  I am trying to loop through various sentences and want to highlight sentences based on a category: 

```
import fitz

doc = fitz.open("output2.pdf")
inst_counter = 0
for i in range(len(sentec_lists)): 

    for j in (range(len(sentec_lists[i]))):
        for pi in range(doc.pageCount):
            page = doc[pi]

            text = str(sentec_lists[i][j])
            text_instances = page.searchFor(text)
            
            for inst in text_instances:
                inst_counter += 1
                highlight = page.addHighlightAnnot(inst)
                highlight.setColors({"stroke":(0, 0, 1), "fill": fills[i]})
        
doc.save("output6.pdf", garbage=4, deflate=True, clean=True)
```

when I run this it gives me an warning: warning: annot type has no fill color
fills is from this code:

```
gray = (0.4392156862745098, 0.4392156862745098, 0.4392156862745098)
blue = (0.52, 0.807843137254902, 0.9803921568627451)
yellow = (0.75, 0.8, 0.95)
pink = (1.0, 0.75, 0.79)
fills= [pink, blue, yellow, gray]

from fitz.utils import getColorList
cl = getColorList()
from fitz.utils import getColor

getColor("gray1")

```
for some reason only yellow works!

**Comment by JorjMcKie** [comment]
Text marker annotations have no fill color by **_PDF specification_**. This is not a (Py-) MuPDF restriction or whatever.
I understand that this may be confusing for highlightings, as obviously an area is being filled - but nevertheless the (border) color is used for this, too.
So change the normal color to your liking and you'll be fine.
Of course you must ``update()`` the annot for this ...

**Reply by rishkum** [reply]
Thanks for the reply, do you mean change the code to :
`                highlight.setColors({"stroke":(0, 0, 1), "color":fills[i]})
                highlight.update()`

from [this](https://stackoverflow.com/questions/60557977/how-to-change-the-highlight-color-in-pdf-using-fitz-module-in-python) answer it looked like highlight.setColors takes a tuple of stroke and fill

**Comment by JorjMcKie** [comment, marked as answer]
Do this:
```python
annot=page.addHihlightAmmot(...)
annot.set_colors(stroke=(0,0,1))
annot.update()
```

**Reply by rishkum** [reply]
Thanks for the reply I got this error: AttributeError: 'Annot' object has no attribute 'set_colors' :(

I tried with both yours(annot=page.addHihlightAmmot(...)) and (annot=page.addHighlightAnnot(...))

setColors() just woks for yellow but when I try gray = (0.4392156862745098, 0.4392156862745098, 0.4392156862745098); nothing shows up. I want to highlight different colors based on different topics

**Reply by JorjMcKie** [reply]
oh dear me, you do not have the current version:
```python
>>> import fitz
>>> help(fitz.Annot.set_colors)
Help on function set_colors in module fitz.fitz:

set_colors(self, colors: Any = None, fill: Any = None, stroke: Any = None) -> None
    Set 'stroke' and 'fill' colors.
    
    Use either a dict or the direct arguments.

>>> 
```
If you don't want or cannot upgrade, use setColors instead.
Here is a snippet, which you can use:
```python
>>> import fitz
>>> doc=fitz.open()
>>> page=doc.new_page()
>>> page.insert_text((100,100), "Highlight this.")
1
>>> rl=page.search_for("highlight this.")
>>> a=page.addHighlightAnnot(rl[0])
>>> a.set_colors(stroke=(0,1,0))
>>> a.update()
>>> doc.save("x.pdf")
```
gives you this:
![grafik](https://user-images.githubusercontent.com/8290722/116436991-f3eddc00-a81a-11eb-937a-3b13d8c756d6.png)

**Reply by rishkum** [reply]
Thanks for this, everything clicks now. Just for anyone coming back to this, the methods use Camel case in my version and not the underscores
